app/src/composables/use-field-tree.ts

```diff
@@ -174,4 +175,20 @@ export function useFieldTree(
 			}
 		}
 	}
+
+	function flattenGroupFields(tree: FieldNode[]) {
+		function flattenGroups(list: FieldNode[]): FieldNode[] {
+			return list.flatMap((item: FieldNode) => {
+				if (Array.isArray(item.children) && item.children.length > 0) {
+					if (item.group === true) {
+						return flattenGroups(item.children);
+					} else {
+						item.children = flattenGroups(item.children);
+					}
+				}
+				return item;
+			});
+		}
+		return flattenGroups(tree);
+	}
```

Comment:
This doesn't seem directly tied to the composable `use-field-tree`. Should we move this to `utils` (and add a unit test in the process)?

Reply:
I was on the fence where this logic should end up. Originally I had this generating the flattened tree and returning it together with the regular field tree here. Since it's a decently abstract function and indeed not directly tied to the `use-field-tree` composable anymore `utils` sounds like the better place for this.